dotnet/src/SemanticKernel/Memory/MemoryRecord.cs

```diff
@@ -110,28 +116,45 @@ public class MemoryRecord : DataEntryBase
     /// Create a memory record from a serialized metadata string.
     /// </summary>
     /// <param name="json">Json string representing a memory record's metadata.</param>
-    /// <param name="embedding">The embedding associated with a memory record.</param>
-    /// <param name="key">Optional existing database key</param>
-    /// <param name="timestamp">optional timestamp</param>
+    /// <param name="embedding">Optional embedding associated with a memory record.</param>
+    /// <param name="key">Optional existing database key.</param>
+    /// <param name="timestamp">optional timestamp.</param>
     /// <returns></returns>
     /// <exception cref="MemoryException"></exception>
     public static MemoryRecord FromJson(
         string json,
-        Embedding<float> embedding,
+        Embedding<float>? embedding,
         string? key = null,
         DateTimeOffset? timestamp = null)
     {
         var metadata = JsonSerializer.Deserialize<MemoryRecordMetadata>(json);
         if (metadata != null)
         {
-            return new MemoryRecord(metadata, embedding, key, timestamp);
+            return new MemoryRecord(metadata, embedding ?? Embedding<float>.Empty, key, timestamp);
         }
 
         throw new MemoryException(
             MemoryException.ErrorCodes.UnableToDeserializeMetadata,
             "Unable to create memory record from serialized metadata");
     }
 
+    /// <summary>
+    /// Create a memory record from a memory record's metadata.
+    /// </summary>
+    /// <param name="metadata">Metadata associated with a memory.</param>
+    /// <param name="embedding">Optional embedding associated with a memory record.</param>
+    /// <param name="key">Optional existing database key.</param>
+    /// <param name="timestamp">optional timestamp.</param>
+    /// <returns></returns>
+    public static MemoryRecord FromMetadata(
+        MemoryRecordMetadata metadata,
+        Embedding<float>? embedding,
+        string? key = null,
+        DateTimeOffset? timestamp = null)
+    {
+        return new MemoryRecord(metadata, embedding ?? Embedding<float>.Empty, key, timestamp);
```

Comment:
Do we want empty embeddings in MemoryRecord? Or should this property be nullable too?

Reply:
I found making the property nullable to be very tedious in terms of null checks EVERYWHERE. An empty embedding serves no purpose and cannot be used for similarity comparisons, so functionally it ensures correct behavior. From my perspective, psychologically, it gives me a better sense that the MemoryRecord was constructed with purpose from something existing as opposed to an accidentally missed field.